vignettes/value-labels.Rmd

```diff
@@ -16,99 +15,139 @@ knitr::opts_chunk$set(
+
+ipumsr uses the `labelled`
+[class](https://haven.tidyverse.org/reference/labelled.html) from the
+[haven](https://haven.tidyverse.org/index.html) package to handle value
+labels.
+
+You can see this in the column data types when loading IPUMS data. Note
+that `<int+lbl>` appears below `STATEFIP`, `ASECFLAG`, and other
+variables:
+
+```{r, message=FALSE}
 library(ipumsr)
+
 ddi <- read_ipums_ddi(ipums_example("cps_00015.xml"))
 cps <- read_ipums_micro(ddi, verbose = FALSE)
 
 cps
 ```
 
-The first clue that some of the variables are labelled is the `<dbl+lbl>`
-that appear below `STATEFIP`, `ASECFLAG` and other variables. The tibble package 
-prints the variable's type information below the variable name, and this "+lbl" indicates
-that the variable uses the `labelled` type. The function `is.labelled()` will also
-tell you if a variable is labelled.
-
-By default the ipumsr package now will attempt to show the value labels when
-printing out a data.frame (like the `"[Wis~"` on STATEFIP above, for Wisconsin). 
-If your console supports colorful output, then the labels will be in 
-a lighter gray than the values. You can turn off this printing behavior by 
-setting the option `options("ipumsr.show_pillar_labels" = FALSE)`)
-
-However, as you can see, there often isn't enough space to see the labels 
-when printing out a full data.frame. So, there are a few options to see the 
-labels directly:
+This indicates that the data contained in these columns are integers but
+include value labels. You can use the function `is.labelled()` to
+determine if a variable is indeed labelled:
+
+```{r}
+is.labelled(cps$STATEFIP)
+```
+
+Some of the labels are actually printed inline alongside their data
+values, but it can be easier to see them by isolating them:
+
 ```{r}
-# Printing the variable directly (or a subset)
+# Labels print when accessing the column
 head(cps$MONTH)
 
-# Just get the labels
+# Get labels alone
 ipums_val_labels(cps$MONTH)
-
-# or if you're working interactively you can use ipums_view
-# ipums_view(ddi)
 ```
 
-# Why use the labelled class instead of base R's factors?
-The usual way to connect numeric data to labels in R is in `factor` variables. 
-Though this data type is more native to R, and more widely supported by R code,
-it was designed for efficient calculations in linear models, not as a general 
-purpose value labeling system and so is missing important features that the 
-value labels provided by IPUMS require.
+## `labelled` vs. `factor`
+
+Base R already supports the linking of numeric data to categories using
+its `factor` data type. While factors may be more familiar, they were
+designed to support efficient calculations in linear models, not as a
+human-readable labeling system for interpreting and processing data.
+
+Compared to factors, `labelled` vectors have a few properties that make
+them more suitable for working with IPUMS data:
+
+-   They provide more complete contextual information about the meaning
```

Comment:
I'm not totally clear about what this means -- my guess is that it's referring to the fact that the values themselves may be meaningful (as in the case of AGE). If that is the point here, we might be able to merge this point with point 3, something like: "They don't require values to be assigned to increasing integers starting at 1, which is important because in some IPUMS variables the values themselves have meaning".